src/NServiceBus.TransportTests/When_failure_happens_after_isolated_send.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace NServiceBus.TransportTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+    using Transport;
+
+    public class When_failure_happens_after_isolated_send : NServiceBusTransportTest
+    {
+        [TestCase(TransportTransactionMode.None)]
+        [TestCase(TransportTransactionMode.ReceiveOnly)]
+        [TestCase(TransportTransactionMode.SendsAtomicWithReceive)]
+        [TestCase(TransportTransactionMode.TransactionScope)]
+        public async Task Should_emit_messages(TransportTransactionMode transactionMode)
+        {
+            var onMessageCalled = new TaskCompletionSource<bool>();
+
+            OnTestTimeout(() => onMessageCalled.SetCanceled());
+
+            await StartPump(async context =>
+                {
+                    if (context.Headers.ContainsKey("CompleteTest"))
+                    {
+                        onMessageCalled.SetResult(false);
+                        return;
+                    }
+
+                    if (context.Headers.ContainsKey("IsolatedSend"))
+                    {
+                        onMessageCalled.SetResult(true);
+                        return;
+                    }
+
+                    await SendMessage(InputQueueName, new Dictionary<string, string>
+                    {
+                        {"IsolatedSend", "true"}
+                    }, context.TransportTransaction, null, DispatchConsistency.Isolated);
+
+                    throw new Exception("Simulated exception");
+                },
+                async context =>
+                {
+                    await SendMessage(InputQueueName, new Dictionary<string, string>
```

Comment:
I'm not sure I understand what this "complete" message does. Why do we need that in the first place? Looks to me like there might be a potential race condition in case this message is handled before the isolated send? (since we always call on error  due to the simulated exception).